diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/behatsdaa.js b/node_modules/israeli-bank-scrapers/lib/scrapers/behatsdaa.js
index a425681..479d207 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/behatsdaa.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/behatsdaa.js
@@ -29,14 +29,15 @@ const PURCHASE_HISTORY_URL = 'https://back.behatsdaa.org.il/api/purchases/purcha
 const debug = (0, _debug.getDebug)('behatsdaa');
 
 function variantToTransaction(variant) {
+  const amount = -variant.customerPrice;
   return {
     type: _transactions.TransactionTypes.Normal,
     identifier: variant.tTransactionID,
     date: (0, _moment.default)(variant.orderDate).format('YYYY-MM-DD'),
     processedDate: (0, _moment.default)(variant.orderDate).format('YYYY-MM-DD'),
-    originalAmount: variant.customerPrice,
+    originalAmount: amount,
     originalCurrency: 'ILS',
-    chargedAmount: variant.customerPrice,
+    chargedAmount: amount,
     chargedCurrency: 'ILS',
     description: variant.name,
     status: _transactions.TransactionStatuses.Completed,
