diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js b/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js
index 09768ac..d839e5e 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js
@@ -259,7 +259,7 @@ class VisaCalScraper extends _baseScraperWithBrowser.BaseScraperWithBrowser {
       const finalMonthToFetchMoment = (0, _moment.default)().add(futureMonthsToScrape, 'month');
       const months = finalMonthToFetchMoment.diff(startMoment, 'months');
       const allMonthsData = [];
-      const frame = _lodash.default.find(frames.result.bankIssuedCards.cardLevelFrames, {
+      const frame = _lodash.default.find(frames.result.bankIssuedCards?.cardLevelFrames, {
         cardUniqueId: card.cardUniqueId
       });
       debug(`fetch pending transactions for card ${card.cardUniqueId}`);
