diff --git a/node_modules/israeli-bank-scrapers/lib/helpers/elements-interactions.js b/node_modules/israeli-bank-scrapers/lib/helpers/elements-interactions.js
index edf5350..df0bce6 100644
--- a/node_modules/israeli-bank-scrapers/lib/helpers/elements-interactions.js
+++ b/node_modules/israeli-bank-scrapers/lib/helpers/elements-interactions.js
@@ -103,7 +103,7 @@ async function pageEval(pageOrFrame, selector, defaultResult, callback, ...args)
     result = await pageOrFrame.$eval(selector, callback, ...args);
   } catch (e) {
     // TODO temporary workaround to puppeteer@1.5.0 which breaks $$eval bevahvior until they will release a new version.
-    if (e.message.indexOf('Error: failed to find elements matching selector') !== 0) {
+    if (e.message.indexOf('Error: failed to find element matching selector') !== 0) {
       throw e;
     }
   }
diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/hapoalim.js b/node_modules/israeli-bank-scrapers/lib/scrapers/hapoalim.js
index 0234816..42128cb 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/hapoalim.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/hapoalim.js
@@ -120,7 +120,7 @@ async function fetchAccountData(page, baseUrl, options) {
     const balanceAndCreditLimitUrl = `${apiSiteUrl}/current-account/composite/balanceAndCreditLimit?accountId=${accountNumber}&view=details&lang=he`;
     const balanceAndCreditLimit = await (0, _fetch.fetchGetWithinPage)(page, balanceAndCreditLimitUrl);
     const balance = balanceAndCreditLimit === null || balanceAndCreditLimit === void 0 ? void 0 : balanceAndCreditLimit.currentBalance;
-    const txnsUrl = `${apiSiteUrl}/current-account/transactions?accountId=${accountNumber}&numItemsPerPage=150&retrievalEndDate=${endDateStr}&retrievalStartDate=${startDateStr}&sortCode=1`;
+    const txnsUrl = `${apiSiteUrl}/current-account/transactions?accountId=${accountNumber}&numItemsPerPage=600&retrievalEndDate=${endDateStr}&retrievalStartDate=${startDateStr}&sortCode=1`;
     const txnsResult = await fetchPoalimXSRFWithinPage(page, txnsUrl, '/current-account/transactions');
     let txns = [];
 
diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js b/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js
index f2566f4..64a0b9f 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js
@@ -199,6 +199,15 @@ async function fetchTransactionsForAccount(page, startDate, accountNumber, scrap
     await Promise.all([page.waitForNavigation({
       waitUntil: 'domcontentloaded'
     }), (0, _elementsInteractions.clickButton)(page, buttonSelector)]);
+    debug('check if month has no transactions');
+    const pageHasNoTransactions = await (0, _elementsInteractions.pageEval)(page, '[id$=FormAreaNoBorder_FormArea_msgboxErrorMessages]', false, element => {
+      const siteValue = (element.innerText || '').replace(/[^ א-ת]/g, '');
+      return siteValue === 'לא נמצאו נתונים';
+    });
+    if (pageHasNoTransactions) {
+      debug('page has no transactions'); // eslint-disable-next-line no-continue
+      continue;
+    }
     debug('find the billing date');
     const billingDateLabel = await (0, _elementsInteractions.pageEval)(page, billingLabelSelector, '', element => {
       return element.innerText;
