diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/base-isracard-amex.js b/node_modules/israeli-bank-scrapers/lib/scrapers/base-isracard-amex.js
index 5f9f3ff..9347392 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/base-isracard-amex.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/base-isracard-amex.js
@@ -191,7 +191,7 @@ async function fetchAllTransactions(page, options, companyServiceOptions, startM
   const results = await Promise.all(allMonths.map(async monthMoment => {
     return fetchTransactions(page, options, companyServiceOptions, startMoment, monthMoment);
   }));
-  const finalResult = options.additionalTransactionInformation ? await getExtraScrap(results, page, companyServiceOptions, allMonths) : results;
+  const finalResult = false && options.additionalTransactionInformation ? await getExtraScrap(results, page, companyServiceOptions, allMonths) : results;
   const combinedTxns = {};
   finalResult.forEach(result => {
     Object.keys(result).forEach(accountNumber => {
@@ -234,6 +234,16 @@ class IsracardAmexBaseScraper extends _baseScraperWithBrowser.BaseScraperWithBro
     });
     await (0, _browser.maskHeadlessUserAgent)(this.page);
     await this.navigateTo(`${this.baseUrl}/personalarea/Login`);
+    try {
+      const initialPageTitle = await this.page.title();
+      debug("initial page title: ", initialPageTitle);
+      if (!initialPageTitle.includes('ישראכרט') && !initialPageTitle.includes('אמריקן אקספרס')) {
+        debug("waiting for page to load");
+        await this.page.waitForNavigation({ timeout: 60_000 });
+      }
+    } catch (error) {
+      debug("error while waiting for navigation: ", error);
+    }
     this.emitProgress(_definitions.ScraperProgressTypes.LoggingIn);
     const validateUrl = `${this.servicesUrl}?reqName=ValidateIdData`;
     const validateRequest = {
diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js b/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js
index b13abfb..72a0be9 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/visa-cal.js
@@ -166,12 +166,12 @@ class VisaCalScraper extends _baseScraperWithBrowser.BaseScraperWithBrowser {
       debug('fetching authorization header');
       const authModule = await (0, _waiting.waitUntil)(async () => {
         const result = await (0, _storage.getFromSessionStorage)(this.page, 'auth-module');
-        return result && result.auth && result.auth.calConnectToken !== null ? result : null;
-      }, 'get authorization header with valid token in session storage', 1000, 50);
-      if (authModule && authModule.auth.calConnectToken !== null) {
+        return result && result.auth && result.auth.calConnectToken && String(result.auth.calConnectToken).trim() ? result : null;
+      }, 'get authorization header with valid token in session storage', 15000, 50);
+      if (authModule && authModule.auth.calConnectToken && String(authModule.auth.calConnectToken).trim()) {
         return `CALAuthScheme ${authModule.auth.calConnectToken}`;
       }
-      throw new Error('could not retrieve authorization header');
+      throw new Error('could not retrieve authorization header' + JSON.stringify(authModule));
     }
     return this.authorization;
   }
@@ -212,7 +212,10 @@ class VisaCalScraper extends _baseScraperWithBrowser.BaseScraperWithBrowser {
             await (0, _elementsInteractions.clickButton)(this.page, 'button.btn-close');
           }
           const request = await this.authRequestPromise;
-          this.authorization = request?.headers()?.authorization;
+          const authorization = request?.headers()?.authorization;
+          if(authorization && String(authorization).trim()) {
+            this.authorization = authorization;
+          }
         } catch (e) {
           const currentUrl = await (0, _navigation.getCurrentUrl)(this.page);
           if (currentUrl.endsWith('dashboard')) return;
