diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/max.js b/node_modules/israeli-bank-scrapers/lib/scrapers/max.js
index 97100e3..cdd56be 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/max.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/max.js
@@ -134,13 +134,13 @@ function getTransactionType(txnTypeStr) {
 
 function getInstallmentsInfo(comments) {
   if (!comments) {
-    return null;
+    return undefined;
   }
 
   const matches = comments.match(/\d+/g);
 
   if (!matches || matches.length < 2) {
-    return null;
+    return undefined;
   }
 
   return {
@@ -153,6 +153,10 @@ function mapTransaction(rawTransaction) {
   const isPending = rawTransaction.paymentDate === null;
   const processedDate = (0, _moment.default)(isPending ? rawTransaction.purchaseDate : rawTransaction.paymentDate).toISOString();
   const status = isPending ? _transactions2.TransactionStatuses.Pending : _transactions2.TransactionStatuses.Completed;
+  const installments = getInstallmentsInfo(rawTransaction.comments);
+  const identifier = installments ?
+    `${rawTransaction.dealData?.arn}_${installments.number}` :
+    rawTransaction.dealData?.arn;
   return {
     type: getTransactionType(rawTransaction.planName),
     date: (0, _moment.default)(rawTransaction.purchaseDate).toISOString(),
@@ -163,7 +167,8 @@ function mapTransaction(rawTransaction) {
     description: rawTransaction.merchantName.trim(),
     memo: rawTransaction.comments,
     category: categories.get(rawTransaction === null || rawTransaction === void 0 ? void 0 : rawTransaction.categoryId),
-    installments: getInstallmentsInfo(rawTransaction.comments) || undefined,
+    installments,
+    identifier,
     status
   };
 }
