diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/behatsdaa.js b/node_modules/israeli-bank-scrapers/lib/scrapers/behatsdaa.js
index a425681..cd25516 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/behatsdaa.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/behatsdaa.js
@@ -29,14 +29,15 @@ const PURCHASE_HISTORY_URL = 'https://back.behatsdaa.org.il/api/purchases/purcha
 const debug = (0, _debug.getDebug)('behatsdaa');
 
 function variantToTransaction(variant) {
+  const amount = -variant.customerPrice
   return {
     type: _transactions.TransactionTypes.Normal,
     identifier: variant.tTransactionID,
     date: (0, _moment.default)(variant.orderDate).format('YYYY-MM-DD'),
     processedDate: (0, _moment.default)(variant.orderDate).format('YYYY-MM-DD'),
-    originalAmount: variant.customerPrice,
+    originalAmount: amount,
     originalCurrency: 'ILS',
-    chargedAmount: variant.customerPrice,
+    chargedAmount: amount,
     chargedCurrency: 'ILS',
     description: variant.name,
     status: _transactions.TransactionStatuses.Completed,
diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/max.js b/node_modules/israeli-bank-scrapers/lib/scrapers/max.js
index 1a4653b..64e562c 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/max.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/max.js
@@ -132,6 +132,7 @@ function getTransactionType(txnTypeStr) {
     case REPLACEMENT_CARD:
     case EARLY_REPAYMENT:
     case MONTHLY_CARD_FEE:
+    case 'חיוב ארנק מטח':
       return _transactions2.TransactionTypes.Normal;
 
     case INSTALLMENTS_TYPE_NAME:
@@ -177,6 +178,17 @@ function getChargedCurrency(currencyId) {
   }
 }
 
+function getMemo(rawTransaction) {
+  let memo = rawTransaction.comments;
+  if (rawTransaction.fundsTransferReceiverOrTransfer) {
+    memo += ` ${rawTransaction.fundsTransferReceiverOrTransfer}`;
+    if(rawTransaction.fundsTransferComment) {
+      memo += `: ${rawTransaction.fundsTransferComment}`;
+    }
+  }
+  return memo;
+}
+
 function mapTransaction(rawTransaction) {
   var _rawTransaction$dealD, _rawTransaction$dealD2;
 
@@ -194,7 +206,7 @@ function mapTransaction(rawTransaction) {
     chargedAmount: -rawTransaction.actualPaymentAmount,
     chargedCurrency: getChargedCurrency(rawTransaction.paymentCurrency),
     description: rawTransaction.merchantName.trim(),
-    memo: rawTransaction.comments,
+    memo: getMemo(rawTransaction),
     category: categories.get(rawTransaction === null || rawTransaction === void 0 ? void 0 : rawTransaction.categoryId),
     installments,
     identifier,
