diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/base-scraper-with-browser.js b/node_modules/israeli-bank-scrapers/lib/scrapers/base-scraper-with-browser.js
index b07269d..b7fe457 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/base-scraper-with-browser.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/base-scraper-with-browser.js
@@ -185,14 +185,14 @@ class BaseScraperWithBrowser extends _baseScraper.BaseScraper {
     });
   }
 
-  async navigateTo(url, page, timeout) {
+  async navigateTo(url, page, timeout, waitUntil = 'load') {
     const pageToUse = page || this.page;
 
     if (!pageToUse) {
       return;
     }
 
-    const options = _objectSpread({}, timeout === null ? null : {
+    const options = _objectSpread({ waitUntil }, timeout === null ? null : {
       timeout
     });
 
@@ -237,7 +237,7 @@ class BaseScraperWithBrowser extends _baseScraper.BaseScraper {
     }
 
     debug('navigate to login url');
-    await this.navigateTo(loginOptions.loginUrl);
+    await this.navigateTo(loginOptions.loginUrl, undefined, undefined, loginOptions.waitUntil);
 
     if (loginOptions.checkReadiness) {
       debug('execute \'checkReadiness\' interceptor provided in login options');
diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/max.js b/node_modules/israeli-bank-scrapers/lib/scrapers/max.js
index 227ce7f..9211478 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/max.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/max.js
@@ -292,7 +292,8 @@ class MaxScraper extends _baseScraperWithBrowser.BaseScraperWithBrowser {
         await (0, _elementsInteractions.waitUntilElementFound)(this.page, '.personal-area > a.go-to-personal-area', true);
       },
       postAction: async () => redirectOrDialog(this.page),
-      possibleResults: getPossibleLoginResults(this.page)
+      possibleResults: getPossibleLoginResults(this.page),
+      waitUntil: 'domcontentloaded'
     };
   }
 
