diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/base-isracard-amex.js b/node_modules/israeli-bank-scrapers/lib/scrapers/base-isracard-amex.js
index 8827ffc..8b9220c 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/base-isracard-amex.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/base-isracard-amex.js
@@ -241,6 +241,16 @@ class IsracardAmexBaseScraper extends _baseScraperWithBrowser.BaseScraperWithBro
     });
     await (0, _browser.maskHeadlessUserAgent)(this.page);
     await this.navigateTo(`${this.baseUrl}/personalarea/Login`);
+    try {
+      const initialPageTitle = await this.page.title();
+      debug("initial page title: ", initialPageTitle);
+      if (!initialPageTitle.includes('ישראכרט') && !initialPageTitle.includes('אמריקן אקספרס')) {
+        debug("waiting for page to load");
+        await this.page.waitForNavigation({ timeout: 60_000 });
+      }
+    } catch (error) {
+      debug("error while waiting for navigation: ", error);
+    }
     this.emitProgress(_definitions.ScraperProgressTypes.LoggingIn);
     const validateUrl = `${this.servicesUrl}?reqName=ValidateIdData`;
     const validateRequest = {
diff --git a/node_modules/israeli-bank-scrapers/lib/scrapers/mizrahi.js b/node_modules/israeli-bank-scrapers/lib/scrapers/mizrahi.js
index b3b708f..32deb0b 100644
--- a/node_modules/israeli-bank-scrapers/lib/scrapers/mizrahi.js
+++ b/node_modules/israeli-bank-scrapers/lib/scrapers/mizrahi.js
@@ -143,7 +143,7 @@ async function extractPendingTransactions(page) {
   });
   return pendingTxn.map(([dateStr, description, incomeAmountStr, amountStr]) => ({
     date: (0, _moment.default)(dateStr, 'DD/MM/YY').toISOString(),
-    amount: parseInt(amountStr, 10),
+    amount: parseFloat(amountStr.replaceAll(",", "")),
     description,
     incomeAmountStr // TODO: handle incomeAmountStr once we know the sign of it
   })).filter(txn => txn.date).map(({
