diff --git a/node_modules/ky/distribution/core/Ky.js b/node_modules/ky/distribution/core/Ky.js
index b37cacc..1b09459 100644
--- a/node_modules/ky/distribution/core/Ky.js
+++ b/node_modules/ky/distribution/core/Ky.js
@@ -150,7 +150,7 @@ export class Ky {
             prefixUrl: String(options.prefixUrl || ''),
             retry: normalizeRetryOptions(options.retry),
             throwHttpErrors: options.throwHttpErrors ?? true,
-            timeout: options.timeout ?? 10_000,
+            timeout: options.timeout ?? 180_000,
             fetch: options.fetch ?? globalThis.fetch.bind(globalThis),
             context: options.context ?? {},
         };
