diff --git a/node_modules/ky/distribution/core/Ky.js b/node_modules/ky/distribution/core/Ky.js
index 1f71af8..ad80643 100644
--- a/node_modules/ky/distribution/core/Ky.js
+++ b/node_modules/ky/distribution/core/Ky.js
@@ -279,7 +279,7 @@ export class Ky {
             prefix: String(options.prefix || ''),
             retry: normalizeRetryOptions(options.retry),
             throwHttpErrors: options.throwHttpErrors ?? true,
-            timeout: options.timeout ?? 10_000,
+            timeout: options.timeout ?? 180_000,
             totalTimeout: options.totalTimeout ?? false,
             fetch: options.fetch ?? globalThis.fetch.bind(globalThis),
             context: options.context ?? {},
